LibreHardwareMonitorLib/Hardware/Cpu/Amd17Cpu.cs

```diff
@@ -13,16 +13,20 @@ namespace LibreHardwareMonitor.Hardware.CPU
     internal sealed class Amd17Cpu : AmdCpu
     {
         private readonly Processor _processor;
-        private int _sensorClock;
-        private int _sensorMulti;
-        private int _sensorPower;
-
-        // counter, to create sensor index values
-        private int _sensorTemperatures;
-        private int _sensorVoltage;
+        private Dictionary<SensorType, int> _sensorTypeIndex;
+        private RyzenSMU _smu;
 
         public Amd17Cpu(int processorIndex, CpuId[][] cpuId, ISettings settings) : base(processorIndex, cpuId, settings)
         {
+            _sensorTypeIndex = new Dictionary<SensorType, int>();
+            foreach (SensorType type in Enum.GetValues(typeof(SensorType)))
+            {
+                _sensorTypeIndex.Add(type, 0);
+            }
```

Comment:
Maybe just initialize this dictionary with the sensor types that are actually used (can be on line 16)? Some of the sensor types aren't used by CPUs.

Reply:
Extracted to a method.
There are so many types of sensors on AMD CPUs, just initialize all to make sure no one is missed. :)